### DISCUSSION TITLE
How to mint NFTs in any collection with my own minting contract?

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#501] [Questions]
### Describe your question in detail.

I am going to mint NFTs in every collection via my own contract. The contract is completed so I can mint tokens in collection that only I created with my contract.(NFT minting dapp template)
So as next step, I am going to mint for every collection even not created by my contract.
How should I do?
First of all, I cannot read the collection info using `useGetCollectionData` in hooks, I think it might be reasoned of this in `getActiveOrNextMintStage.ts`

```
export const getActiveOrNextMintStage = async ({
  collection_address,
}: GetRegistryArguments) => {
  const mintStageRes = await aptosClient().view<[{ vec: [string] | [] }]>({
    payload: {
      function: `${AccountAddress.from(
        MODULE_ADDRESS
      )}::LaunchB0xx::get_active_or_next_mint_stage`,
      functionArguments: [collection_address],
    },
  });
  return mintStageRes[0].vec;
};
```

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xbrw** [comment]
This is hard to debug without looking at your source code. Could the function you are trying to use `get_active_or_next_mint_stage ` specific to the collections created by your contract? This could explain why you're having trouble retrieving data outside the scope of your contract.

**Reply by CryptainGolD** [reply]
Sorry, this is the one I used
```
import { aptosClient } from "../utils/aptosClient";
import { MODULE_ADDRESS } from "../config";

export type MintNftArguments = {
  account: any,
  collectionId: string;
  amount: number;
};

export const mintNFT = async (args: MintNftArguments) => {
  const { account, collectionId, amount } = args;
  const transaction = await aptosClient().transaction.build.simple({
    sender: account.accountAddress,
    data: {
      function: `${MODULE_ADDRESS}::LaunchB0xx::mint_nft`,
      functionArguments: [collectionId, amount],
    },
  });

  const response = await aptosClient().signAndSubmitTransaction({
    signer: account,
    transaction,
  });

  await aptosClient().waitForTransaction({
    transactionHash: response.hash,
  });
  
  return response.hash
};
```
This is based on nft-minting-dapp template

I am going to mint nft once I input collection id, and amount, and private key.
🙏 🙏 🙏
